brainframe/cli/commands/self_update.py

```diff
@@ -63,18 +65,21 @@ def self_update():
             error_message=response.text,
         )
 
-    # Overwrite the existing executable with the new one
-    os.remove(executable_path)
-    with executable_path.open("wb") as file_:
+    with NamedTemporaryFile("wb") as new_executable:
         for block in response.iter_content(_BLOCK_SIZE):
-            file_.write(block)
+            new_executable.write(block)
+        new_executable.flush()
+
+        # Set the result as executable
+        current_stat = os.stat(executable_path)
+        os.chmod(
+            executable_path,
+            current_stat.st_mode | stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH,
+        )
```

Comment:
You should be able to use [`Path.chmod`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.chmod) instead of using the `os` library here